pallets/loans/src/entities/loans.rs

```diff
@@ -345,6 +361,18 @@ impl<T: Config> ActiveLoan<T> {
 			Error::<T>::from(BorrowLoanError::MaturityDatePassed)
 		);
 
+		let expected_payment = self.schedule.expected_payment(
+			self.origination_date,
```

Comment:
I think we should generate cashflows here from `repayments_on_schedule_until` rather than from `origination_date`. Since otherwise, if there is an additional borrow, the historic repayments before today would have also changed, and suddenly there will be payments overdue that weren't overdue before. That was the goal with the `repayments_on_schedule_until` variable.

Reply:
That's right! Good catch 👍🏻